tests/mir-opt/inline/unwrap_unchecked.unwrap_unchecked.PreCodegen.after.mir

```diff
@@ -5,8 +5,7 @@ fn unwrap_unchecked(_1: Option<T>) -> T {
     let mut _0: T;                       // return place in scope 0 at $DIR/unwrap_unchecked.rs:+0:54: +0:55
     scope 1 (inlined #[track_caller] Option::<T>::unwrap_unchecked) { // at $DIR/unwrap_unchecked.rs:10:9: 10:27
         debug self => _1;                // in scope 1 at $SRC_DIR/core/src/option.rs:LL:COL
-        let mut _2: &std::option::Option<T>; // in scope 1 at $SRC_DIR/core/src/option.rs:LL:COL
```

Comment:
This file is a nice demonstration: the function had a stray `&Option` that got inlined from `unwrap_unchecked` that was never calculated, but still had a `StorageLive`+`StorageDead` because it was the `self` to `Option::is_some`.

Now it's removed as irrelevant, as seems entirely reasonable to me.